lib/utils/verify-signatures.js

```diff
@@ -0,0 +1,460 @@
+      const workspace = edge
+        && edge.to
+        && edge.to.target
+        && edge.to.target.isWorkspace
+
+      if (workspace) {
+        this.getEdgesOut(edge.to.target)
+      }
+    }
+  }
+
+  async getPackument (spec) {
+    const packument = await pacote.packument(spec, {
+      ...this.npm.flatOptions,
+      fullMetadata: this.npm.config.get('long'),
+      preferOffline: true,
+    })
+    return packument
+  }
+
+  async getKeys ({ registry }) {
+    const cachePath = path.join(this.npm.cache, '_cacache')
+    const cachedKey = `${registry}-/npm/v1/keys`
+
+    try {
+      const entry = await cacache.get(cachePath, cachedKey)
+      const cache = jsonParse(entry.data)
+      if (cache.expires && Date.now() > Date.parse(cache.expires)) {
+        await cacache.rm.entry(cachePath, cachedKey)
+        throw new Error('Cache expired')
+      } else {
+        return cache.keys
+      }
+    } catch {
+      const keys = await fetch.json('/-/npm/v1/keys', {
+        ...this.npm.flatOptions,
+        registry,
+      }).then(({ keys }) => keys.map((key) => {
+        key.pemKey = `-----BEGIN PUBLIC KEY-----\n${key.key}\n-----END PUBLIC KEY-----`
+        return key
+      })).catch(err => {
+        if (err.code === 'E404') {
+          return null
+        } else {
+          throw err
+        }
+      })
+      const inOneWeekMs = 1000 * 60 * 60 * 24 * 7
+      const cache = {
+        expires: new Date(Date.now() + inOneWeekMs).toISOString(),
+        keys,
+      }
+      await cacache.put(cachePath, cachedKey, JSON.stringify(cache))
+      return keys
+    }
+  }
+
+  async getVerifiedInfo (edge) {
+    let alias = false
+    try {
+      alias = npa(edge.spec).subSpec
+    } catch (err) {
+    }
+    const spec = npa(alias ? alias.name : edge.name)
+    const node = edge.to || edge
+    const { path, location } = node
+    const { version } = node.package || {}
+    if (!version) {
+      return
+    }
+    const type = edge.optional ? 'optionalDependencies'
+      : edge.peer ? 'peerDependencies'
+      : edge.dev ? 'devDependencies'
+      : 'dependencies'
+
+    for (const omitType of this.npm.config.get('omit')) {
+      if (node[omitType]) {
+        return
+      }
+    }
+
+    // deps different from prod not currently
+    // on disk are not included in the output
+    if (edge.error === 'MISSING' && type !== 'dependencies') {
+      return
+    }
+
+    try {
+      const packument = await this.getPackument(spec)
+      // if it's not a range, version, or tag, skip it
+      try {
+        if (!npa(`${edge.name}@${edge.spec}`).registry) {
+          return null
+        }
+      } catch (err) {
+        return null
+      }
+
+      const name = alias ? edge.spec.replace('npm', edge.name) : edge.name
+      const { homepage } = packument
+
+      const registry = fetch.pickRegistry(spec, this.npm.flatOptions)
+
+      const versionPackument = pickManifest(packument, version, this.npm.flatOptions)
```

Comment:
I think we can just use `pacote.manifest` with the spec and not bother with all this?

Reply:
The failing here is in `pacote.manifest` losing the time info.  There's no reason we couldn't attach this to the manifest like we do with `_from` and `_integrity`